bin/common.sh

```diff
@@ -38,17 +38,17 @@ function abort() {
 }
 trap abort SIGINT
 
-#######################################
-# https://github.com/google/styleguide/blob/gh-pages/shellguide.md#stdout-vs-stderr
-#######################################
+#####################################################################################
+# https://github.com/google/styleguide/blob/gh-pages/shellguide.md#stdout-vs-stderr #
+#####################################################################################
 function err() {
   echo "[$(date +'%Y-%m-%dT %T.%3N')] ERROR: $*" >&2
 }
 
 function _rind() {
   local cmd="$1"
   shift
-  if [ $has_docker = 'true' ] && [ -n "$IN_DOCKER" ]; then
+  if [ $has_docker = 'true' ] && [[ ! $IN_DOCKER ]]; then
```

Comment:
`[[` is more powerful (https://stackoverflow.com/a/49825114).

Also discovered that the `[` check doesn't always work properly on Linux hosts.

Reply:
Read that and I favor single brackets as they are more DRY and should just work

Reply:
Let's override the style guide in that case. Do you agree with the style guide then that in the following cases `[[` is preferred:
- pathname expansion
- word splitting
- regex matching, e.g: `[ ab = a? ]: a? glob expands. So may be true or false depending on the files in the current directory.` -> is unreliable, so have to use `[[ ab = a? ]]: true, because it does pattern matching (* ? [ are magic). Does not glob expand to files in current directory.` 
- `[[ ab =~ ab? ]]` because `[ a =~ a ]: syntax error. No bash equivalent.`
- logical AND-s, e.g. `[[ a = a && b = b ]]` (or do we prefer `[ a=a ] && [ b = b]`?)
- `x='a b'; [ $x = 'a b' ]: syntax error, expands to [ a b = 'a b' ]` so we have to use `x='a b'; [[ $x = 'a b' ]]: true, quotes not needed`

`[` in all other cases.

Reply:
Also in our conversation today we wanted to keep the Bash version consistent, here is mine:

```
bash --version
GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

If you can send me yours then I will install that one and we check for something like `[ $(bash -c --version) = vX.Y.Z.`

Reply:
```sh
bash --version
GNU bash, version 5.1.4(1)-release (x86_64-apple-darwin20.2.0)
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
```

Reply:
We just settle for the one that is most prevalent. If that is 5 or 4 I don't care. No need to show you mine ;p